account-kit/smart-contracts/src/ma-v2/client/client.test.ts

```diff
@@ -182,17 +192,181 @@ describe("MA v2 Tests", async () => {
     ).rejects.toThrowError();
   });
 
+  it("installs paymaster guard module, verifies use of valid paymaster, then uninstalls module", async () => {
+    let provider = (
+      await givenConnectedProvider({
+        signer,
+        usePaymaster: true,
+      })
+    ).extend(installValidationActions);
+
+    await setBalance(client, {
+      address: provider.getAddress(),
+      value: parseEther("2"),
+    });
+
+    const paymaster = paymaster070.getPaymasterStubData();
+
+    const hookInstallData = PaymasterGuardModule.encodeOnInstallData({
+      entityId: 0,
+      paymaster: "paymaster" in paymaster ? paymaster.paymaster : "0x0", // dummy value for paymaster address if it DNE
+    });
+
+    const installResult = await provider.installValidation({
+      validationConfig: {
+        moduleAddress: zeroAddress,
+        entityId: 0,
+        isGlobal: true,
+        isSignatureValidation: true,
+        isUserOpValidation: true,
+      },
+      selectors: [],
+      installData: "0x",
+      hooks: [
+        {
+          hookConfig: {
+            address: getDefaultPaymasterGuardModuleAddress(provider.chain),
+            entityId: 0, // uint32
+            hookType: HookType.VALIDATION,
+            hasPreHooks: true,
+            hasPostHooks: true,
+          },
+          initData: hookInstallData,
+        },
+      ],
+    });
+
+    // verify hook installtion succeeded
+    await expect(
+      provider.waitForUserOperationTransaction(installResult)
+    ).resolves.not.toThrowError();
```

Comment:
This is fine, but note that `.resolves.not.toThrowError()` isn't particularly better than just awaiting the promise, which will also fail the test if the promise fails, e.g.
```ts
await provider.waitForUserOperationTransaction(
  installResult,
);
```